File: flink-ai-flow/ai_flow/model_center/entity/model_version.py
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#
from ai_flow.model_center.entity._model_repo_entity import _ModelRepoEntity


class ModelVersion(_ModelRepoEntity):
    """
    AIFlow entity for Model Version. A model version is uniquely identified using underlying
    :py:class:`ai_flow.model_center.entity.RegisteredModel` and user-defined version.
    """

    def __init__(self, model_name, model_version):
        """
        Construct a :py:class:`ai_flow.model_center.entity.ModelVersion` instance

        :param model_name: Unique name for this registered model within Model Registry.
        :param model_version: String user-defined model version
        """
        super(ModelVersion, self).__init__()
        self._model_name = model_name
        self._model_version = model_version

    @property
    def model_name(self):
        """Name for registered model"""
        return self._model_name

    @property
    def model_version(self):
        """String user-defined version"""
        return self._model_version

    # proto mappers
    @classmethod
    def from_proto(cls, proto):
        model_meta = proto.model_meta
        return ModelVersion(model_meta.model_name.value if model_meta.HasField("model_name") else None,
                            model_meta.model_version.value if model_meta.HasField("model_version") else None)

    @classmethod
    def from_resp_proto(cls, proto):
        return ModelVersion(proto.model_name.value if proto.HasField("model_name") else None,
                            proto.model_version.value if proto.HasField("model_version") else None)

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/entity/ModelVersion.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package com.aiflow.entity;

import com.aiflow.common.ModelStage;
import com.aiflow.proto.Message.ModelMetaParam;
import com.aiflow.proto.Message.ModelVersionMeta;
import com.aiflow.proto.Message.ModelVersionStatus;

public class ModelVersion {

    private String modelName;
    private String modelVersion;
    private String modelPath;
    private String modelType;
    private String versionDesc;
    private ModelVersionStatus versionStatus;
    private ModelStage currentStage;

    public ModelVersion() {
    }

    public ModelVersion(String modelName, String modelVersion) {
        this.modelName = modelName;
        this.modelVersion = modelVersion;
    }

    public ModelVersion(String modelName, String modelVersion, String modelPath, String modelType, String versionDesc, ModelVersionStatus versionStatus, ModelStage currentStage) {
        this.modelName = modelName;
        this.modelVersion = modelVersion;
        this.modelPath = modelPath;
        this.modelType = modelType;
        this.versionDesc = versionDesc;
        this.versionStatus = versionStatus;
        this.currentStage = currentStage;
    }

    public String getModelName() {
        return modelName;
    }

    public void setModelName(String modelName) {
        this.modelName = modelName;
    }

    public String getModelVersion() {
        return modelVersion;
    }

    public void setModelVersion(String modelVersion) {
        this.modelVersion = modelVersion;
    }

    public String getModelPath() {
        return modelPath;
    }

    public void setModelPath(String modelPath) {
        this.modelPath = modelPath;
    }

    public String getModelType() {
        return modelType;
    }

    public void setModelType(String modelType) {
        this.modelType = modelType;
    }

    public String getVersionDesc() {
        return versionDesc;
    }

    public void setVersionDesc(String versionDesc) {
        this.versionDesc = versionDesc;
    }

    public ModelVersionStatus getVersionStatus() {
        return versionStatus;
    }

    public void setVersionStatus(ModelVersionStatus versionStatus) {
        this.versionStatus = versionStatus;
    }

    public ModelStage getCurrentStage() {
        return currentStage;
    }

    public void setCurrentStage(ModelStage currentStage) {
        this.currentStage = currentStage;
    }

    @Override
    public String toString() {
        return "ModelVersion{" +
                "modelName='" + modelName + '\'' +
                ", modelVersion='" + modelVersion + '\'' +
                ", modelPath='" + modelPath + '\'' +
                ", modelType='" + modelType + '\'' +
                ", versionDesc='" + versionDesc + '\'' +
                ", versionStatus=" + versionStatus +
                ", currentStage=" + currentStage +
                '}';
    }

    public static ModelVersion buildModelVersion(ModelMetaParam modelMetaParam) {
        return modelMetaParam == null ? null : new ModelVersion(modelMetaParam.getModelName().getValue(),
                modelMetaParam.getModelVersion().getValue());
    }

    public static ModelVersion buildModelVersion(ModelVersionMeta modelVersionMeta) {
        return modelVersionMeta == null ? null : new ModelVersion(modelVersionMeta.getModelName(),
                modelVersionMeta.getModelVersion(), modelVersionMeta.getModelPath().getValue(),
                modelVersionMeta.getModelType().getValue(), modelVersionMeta.getVersionDesc().getValue(),
                modelVersionMeta.getVersionStatus(), ModelStage.getModelStage(modelVersionMeta.getCurrentStage()));
    }
}